FWCore/Framework/src/Worker.cc

```diff
@@ -395,10 +395,11 @@ namespace edm {
 
   void Worker::runAcquire(EventTransitionInfo const& info,
                           ParentContext const& parentContext,
-                          WaitingTaskWithArenaHolder& holder) {
+                          WaitingTaskHolder& holder) {
     ModuleContextSentry moduleContextSentry(&moduleCallingContext_, parentContext);
     try {
-      convertException::wrap([&]() { this->implDoAcquire(info, &moduleCallingContext_, holder); });
+      WaitingTaskWithArenaHolder holderWithArena{holder};
+      convertException::wrap([&]() { this->implDoAcquire(info, &moduleCallingContext_, holderWithArena); });
```

Comment:
The `runAcquire()` is called by `runAcquireAfterAsyncPrefetch()` that owns the `holder` beyond the scope of `runAcquire()` function. Therefore the destructor or `doneWaiting()` of this `holderWithArena` should not lead to the task reference count to drop to 0.

It could be cleaner to percolate the `holder` as `WaitingTaskHolder` through the `implDoAcquire()` functions, but that seemed a little bit tedious. I could do it nevertheless, but I wanted to get feedback on this PR in general first.